packages/caravan-fees/src/rbf.ts

```diff
@@ -0,0 +1,448 @@
+    return this.newPsbt;
+  }
+
+  /**
+   * Prepares a cancellation replacement transaction
+   *
+   * This method creates a new PSBT that attempts to cancel the original transaction
+   * by redirecting all funds (minus the new fee) to a specified address.
+   *
+   * @param {string} destinationAddress - The address to send funds to in the cancellation transaction
+   * @returns {PsbtV2} A new PSBT representing the cancellation transaction
+   * @throws {Error} If unable to create a valid cancellation transaction
+   */
+  public prepareCanceled(destinationAddress: string): PsbtV2 {
+    this.updateSequenceNumbers();
+    this.createCancelOutput(destinationAddress);
+    this.updateFeeRate();
+    return this.newPsbt;
+  }
+
+  /**
+   * Updates sequence numbers for all inputs
+   *
+   * This method ensures that all inputs have a sequence number that signals RBF,
+   * as required by BIP125.
+   *
+   * @private
+   */
+  private updateSequenceNumbers(): void {
+    for (let i = 0; i < this.newPsbt.PSBT_GLOBAL_INPUT_COUNT; i++) {
+      const currentSequence = this.newPsbt.PSBT_IN_SEQUENCE[i];
+      if (currentSequence === null || currentSequence >= 0xfffffffe) {
+        this.newPsbt.PSBT_IN_SEQUENCE[i] = RBF_SEQUENCE;
+      }
+    }
+  }
+
+  /**
+   * Adjusts transaction outputs to accommodate the increased fee
+   *
+   * This method implements the logic for subtracting the fee increase from outputs,
+   * either from a specific output or proportionally from all non-change outputs.
+   *
+   * @private
+   * @throws {Error} If fee subtraction would create dust outputs
+   */
+  private adjustOutputs(): void {
+    const feeIncrease = this.calculateFeeIncrease();
+    if (this.options.subtractFeeFromOutput !== undefined) {
+      this.subtractFeeFromSpecificOutput(feeIncrease);
+    } else {
+      this.subtractFeeProportionally(feeIncrease);
+    }
+  }
+
+  private subtractFeeFromSpecificOutput(feeIncrease: BigNumber): void {
+    const outputIndex = this.options.subtractFeeFromOutput!;
+    const output = this.newPsbt.PSBT_OUT_AMOUNT[outputIndex];
+    if (output === undefined) {
+      throw new Error(`Output at index ${outputIndex} does not exist`);
+    }
+    const outputAmount = new BigNumber(output.toString());
+    const newAmount = BigNumber.max(
+      outputAmount.minus(feeIncrease),
+      new BigNumber(0),
+    );
+    if (newAmount.isLessThan(this.options.dustThreshold!)) {
+      throw new Error(
+        `Output ${outputIndex} would become dust after fee subtraction`,
+      );
+    }
+    // Create a new output with the updated amount
+    const newOutput = {
+      amount: Number(newAmount.toString()),
+      script: Buffer.from(this.newPsbt.PSBT_OUT_SCRIPT[outputIndex], "hex"),
+    };
+
+    // Remove the old output and add the new one
+    this.newPsbt.deleteOutput(outputIndex);
+    this.newPsbt.addOutput(newOutput);
+  }
+
+  private subtractFeeProportionally(feeIncrease: BigNumber): void {
```

Comment:
Could you add some documentation about this? I'm guessing it's from all outputs right?

Reply:
So for the outputs we are only subtracting to fee bump from the non-change outputs ...
I have added this documentation for both the methods in the latest commit 

*for :*  ```subtractFeeProportionally```
```js
  /**
   * Subtracts the fee proportionally from all non-change outputs
   *
   * This method reduces the amount of all non-change outputs proportionally
   * to cover the increased fee. If any resulting output would be dust, it throws an error.
   *
   * @private
   * @throws {Error} If any output would become dust after fee subtraction
   */
```
*for :*  ```subtractFeeFromSpecificOutput```

```js
  /**
   * Subtracts the fee from a specific output
   *
   * This method reduces the amount of a specific output to cover the increased fee.
   * If the resulting output would be dust, it throws an error.
   *
   * @private
   * @throws {Error} If the specified output would become dust after fee subtraction
   */
```